### PR TITLE
MGMT-3491 swagger changes for additional OLM based services

### PR DESCRIPTION
**Pull request by gamli75**
@avishayt @ronniel1 @filanov please review

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -2521,6 +2523,22 @@ paths:
...
+    get:
+      tags:
+        - operators
+      operationId: ListSupportedOperators
+      description: Retrieves the list of supported operators
+      responses:
+        "200":
+          description: Success.
+          schema:
+            $ref: '#/definitions/list-operators'
```
Comment (avishayt) [inline swagger.yaml:2536]: I think this should return a list of strings (e.g., ["lso", "ocs", "..."]?

We also wanted another API to get the properties JSON for each operator:
GET /supported-operators/{operator_type}

File: swagger.yaml [hunk 2]
```diff
@@ -2521,6 +2523,22 @@ paths:
...
+        "503":
```
Comment (avishayt) [inline swagger.yaml:2537]: Don't need 503, but need 401, 403, 500.

File: swagger.yaml [hunk 3]
```diff
@@ -2521,6 +2523,22 @@ paths:
           schema:
             $ref: '#/definitions/error'
 
+  /supported_operators:
```
Comment (avishayt) [inline swagger.yaml:2526]:
```suggestion
  /supported-operators:
```

File: swagger.yaml [hunk 4]
```diff
@@ -25,6 +25,8 @@ tags:
     description: Manifests for customizing a cluster installation.
   - name: versions
     description: Information regarding versions.
+  - name: operators
```
Comment (avishayt) [inline swagger.yaml:28]: The linter should require alphabetical order here?

File: swagger.yaml [hunk 5]
```diff
@@ -4214,6 +4237,19 @@ definitions:
       - 'all'
       - ''
 
+  operator-type:
+    type: string
+    enum:
+      - 'lso'
+      - 'ocs'
+
+  list-operators:
```
Comment (avishayt) [inline swagger.yaml:4246]: You are allowing users to set the status/status_info in create/update cluster

Comment (gamli75) [reply swagger.yaml:4308]: fixed

File: swagger.yaml (hunk 4 above)
Comment (gamli75) [reply swagger.yaml:26]: the linter didn't fail but I fixed it

File: swagger.yaml (hunk 3 above)
Comment (gamli75) [reply swagger.yaml]: fixed

File: swagger.yaml (hunk 2 above)
Comment (gamli75) [reply swagger.yaml]: fixed

File: swagger.yaml (hunk 1 above)
Comment (gamli75) [reply swagger.yaml]: fixed

File: swagger.yaml [hunk 6]
```diff
@@ -2521,6 +2523,67 @@ paths:
...
+      tags:
+        - operators
+      security:
+        - userAuth: [admin, read-only-admin, user]
+      description: Lists properties for an operator type.
+      operationId: ListOpertorProperties
+      parameters:
+        - in: path
+          name: operator_type
+          description: The operatort type.
```
Comment (avishayt) [inline swagger.yaml:2561]:
```suggestion
          description: The operator type.
```

**Review by avishayt — COMMENTED** [review]
Small typo, otherwise looks good

File: swagger.yaml (hunk 6 above)
Comment (gamli75) [reply swagger.yaml]: fixed

**Review by avishayt — APPROVED** [review]

**Comment by priyanka19-98** [comment]
@gamli75, the generic validation function (where the validations of all operators will take place), is required to be part of this PR, or it will come with ocs PR?

File: swagger.yaml [hunk 7]
```diff
@@ -2521,6 +2523,67 @@ paths:
...
+            $ref: '#/definitions/error'
+
+  /supported-operators/{operator_type}:
+    get:
+      tags:
+        - operators
+      security:
+        - userAuth: [admin, read-only-admin, user]
+      description: Lists properties for an operator type.
+      operationId: ListOpertorProperties
```
Comment (nbalacha) [inline swagger.yaml:2557]: Nit: typo.Should be ListOperatorProperties

**Comment by gamli75** [comment]
> @gamli75, the generic validation function (where the validations of all operators will take place), is required to be part of this PR, or it will come with ocs PR?
> https://github.com/openshift/assisted-service/blob/master/internal/cluster/validator.go (all the validations are added here) and https://github.com/openshift/assisted-service/blob/master/internal/cluster/statemachine.go#L94 . Looks good overall

@priyanka19-98 for the first part(this PR) - we will keep the OCS validation as you wrote in your PR. After I'll add the generic interface - we will have a method that returns the validation result. You just need to change the id from: allOperatorsMinRequirementsSatisfied to cso-requirements-satisfied.

Once I'll have the generic interface - I'll make the changes and share the PR with you.

File: internal/ignition/ignition.go [hunk 8]
```diff
@@ -205,7 +205,16 @@ func (g *installerGenerator) UploadToS3(ctx context.Context) error {
 	return uploadToS3(ctx, g.workDir, g.cluster, g.s3Client, g.log)
 }
 func (g *installerGenerator) checkLsoEnabled() bool {
```
Comment (filanov) [inline internal/ignition/ignition.go:207]: why isn't it part of the operator api? `IsEnabled` or even better, why not just call to generate and if it's not enabled then return an empty list or files?

Comment (gamli75) [reply internal/ignition/ignition.go:207]: we will do it in the second part.
We decided to split it into 2 parts: 
1. swagger changes for saving the operators 
2. add the generic interface - I'll do it after Prianka will merge her code

**Review by avishayt — APPROVED** [review]

**Pull request merged**